raiden/network/proxies/token_network.py

```diff
@@ -223,27 +273,44 @@ def detail_participant(
             'nonce': data[4],
         }
 
-    def detail_channel(self, participant1: typing.Address, participant2: typing.Address) -> Dict:
-        """ Returns a dictionary with the channel specific information. """
+    def detail_channel(
+            self,
+            participant1: typing.Address,
+            participant2: typing.Address,
+            channel_identifier: typing.Optional[typing.ChannelID] = None,
+    ) -> Dict[str, typing.Union[typing.ChannelID, typing.BlockNumber, str]]:
+        """ Returns a dictionary with the channel specific information.
+
+        If no specific channel_identifier is given then it tries to see if there
+        is a currently open channel and uses that identifier.
+
+        """
+        channel_identifier = self._inspect_channel_identifier(
+            channel_identifier=channel_identifier,
+            participant1=participant1,
+            participant2=participant2,
+            called_by_fn='detail_channel',
+        )
+
         channel_data = self._call_and_check_result(
             'getChannelInfo',
+            channel_identifier,
             to_checksum_address(participant1),
             to_checksum_address(participant2),
         )
 
-        assert isinstance(channel_data[0], typing.T_ChannelID)
-
         return {
-            'channel_identifier': channel_data[0],
-            'settle_block_number': channel_data[1],
-            'state': channel_data[2],
+            'channel_identifier': channel_identifier,
+            'settle_block_number': channel_data[0],
+            'state': channel_data[1],
```

Comment:
what about giving a name for these constants? preferably in the raiden-contracts repo

Reply:
Sure